components/string.rst

```diff
@@ -0,0 +1,451 @@
+
+    // returns the contents found before/after the first occurrence of the given string
+    u('hello world')->before('world');   // 'hello '
+    u('hello world')->before('o');       // 'hell '
+    u('hello world')->before('o', true); // 'hello'
+
+    u('hello world')->after('hello');   // ' world'
+    u('hello world')->after('o');       // ' world'
+    u('hello world')->after('o', true); // 'o world'
+
+    // returns the contents found before/after the last occurrence of the given string
+    u('hello world')->beforeLast('o');       // 'hello w'
+    u('hello world')->beforeLast('o', true); // 'hello wo'
+
+    u('hello world')->afterLast('o');       // 'rld'
+    u('hello world')->afterLast('o', true); // 'orld'
+
+Methods to Pad and Trim
+~~~~~~~~~~~~~~~~~~~~~~~
+
+::
+
+    // makes a string as long as the first argument by adding the given
+    // string at the beginning, end or both sides of the string
+    u(' Lorem Ipsum ')->padBoth(20, '-'); // '--- Lorem Ipsum ----'
+    u(' Lorem Ipsum')->padStart(20, '-'); // '-------- Lorem Ipsum'
+    u('Lorem Ipsum ')->padEnd(20, '-');   // 'Lorem Ipsum --------'
+
+    // repeats the given string the number of times passed as argument
+    u('_.')->repeat(10); // '_._._._._._._._._._.'
+
+    // removes the given characters (by default, white spaces) from the string
+    u('   Lorem Ipsum   ')->trim(); // 'Lorem Ipsum'
+    u('Lorem Ipsum   ')->trim('m'); // 'Lorem Ipsum   '
+    u('Lorem Ipsum')->trim('m');    // 'Lorem Ipsu'
+
+    u('   Lorem Ipsum   ')->trimStart(); // 'Lorem Ipsum   '
+    u('   Lorem Ipsum   ')->trimEnd();   // '   Lorem Ipsum'
+
+Methods to Search and Replace
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+::
+
+    // checks if the string starts/ends with the given string
+    u('https://symfony.com')->startsWith('https'); // true
+    u('report-1234.pdf')->endsWith('.pdf');        // true
+
+    // checks if the string contents are exactly the same as the given contents
+    u('foo')->equalsTo('foo'); // true
+
+    // checks if the string content match the given regular expression
+    u('avatar-73647.png')->match('/avatar-(\d+)\.png/');
+    // result = ['avatar-73647.png', '73647']
+
+    // finds the position of the first occurrence of the given string
+    // (the second argument is the position where the search starts and negative
+    // values have the same meaning as in PHP functions)
+    u('abcdeabcde')->indexOf('c');     // 2
+    u('abcdeabcde')->indexOf('c', 2);  // 2
+    u('abcdeabcde')->indexOf('c', -4); // 7
+    u('abcdeabcde')->indexOf('eab');   // 4
+    u('abcdeabcde')->indexOf('k');     // null
+
+    // finds the position of the last occurrence of the given string
+    // (the second argument is the position where the search starts and negative
+    // values have the same meaning as in PHP functions)
+    u('abcdeabcde')->indexOfLast('c');     // 7
+    u('abcdeabcde')->indexOfLast('c', 2);  // 7
+    u('abcdeabcde')->indexOfLast('c', -4); // 2
```

Comment:
I think this should be `7`.

Reply:
I've run all code using a script ... but I've just double checked and the result is `2`.